### PR TITLE
feat(ha-node): allow listening on IPv6 Pod IPs

### PR DESCRIPTION
**Pull request by michaelbeaumont**
See https://github.com/openebs/mayastor/issues/1731

### CODE REVIEW

File: control-plane/agents/src/bin/ha/node/main.rs [hunk 1]
```diff
@@ -40,8 +43,14 @@ struct Cli {
     node_name: String,
 
     /// IP address and port for the ha node-agent to listen on.
-    #[clap(short, long, default_value = DEFAULT_NODE_AGENT_SERVER_ADDR)]
-    grpc_endpoint: SocketAddr,
+    #[clap(short, long = "grpc-endpoint")]
```
Comment (tiagolobocastro) [inline control-plane/agents/src/bin/ha/node/main.rs]: Does the short become `d` or `g` in this case?

**Review by tiagolobocastro — COMMENTED** [review]
bors d+

File: control-plane/agents/src/bin/ha/node/main.rs (hunk 1 above)
Comment (michaelbeaumont) [reply control-plane/agents/src/bin/ha/node/main.rs]: Ah good spot, yeah that should be made explicit too

Comment (michaelbeaumont) [reply control-plane/agents/src/bin/ha/node/main.rs]: fixed this

**Comment by michaelbeaumont** [comment]
bors try

**Review by tiagolobocastro — APPROVED** [review]

**Comment by tiagolobocastro** [comment]
hmm odd:
```
[2024-10-09T18:59:54.970Z] FAILED tests/bdd/features/capacity/thin/volume/test_mixed.py::test_outofspace_condition_does_not_affect_thick_volumes_on_mixed_pools - AssertionError: FIO should have failed!

[2024-10-09T18:59:54.970Z] assert 0 != 0
```
bors try

**Review by dsharma-dc — APPROVED** [review]

**Comment by tiagolobocastro** [comment]
This may now be merged @michaelbeaumont ?

**Comment by michaelbeaumont** [comment]
bors merge

**Pull request merged**